src/regexp.ts

```diff
@@ -103,12 +104,27 @@ export class BSONRegExp extends BSONValue {
     throw new BSONError(`Unexpected BSONRegExp EJSON object form: ${JSON.stringify(doc)}`);
   }
 
-  /** @internal */
-  [Symbol.for('nodejs.util.inspect.custom')](): string {
-    return this.inspect();
+  inspect(depth?: number, options?: unknown): string {
+    const stylize = getStylizeFunction(options);
+    const pattern = stylize(`'${this.pattern}'`, 'regexp');
+    const flags = stylize(`'${this.options}'`, 'regexp');
+    return `new BSONRegExp(${pattern}, ${flags})`;
   }
+}
+
+/** @internal */
+type StylizeFunction = (x: string, style: string) => string;
+/** @internal */
+function getStylizeFunction(options?: unknown): StylizeFunction {
+  const stylizeExists =
+    options != null &&
+    typeof options === 'object' &&
+    'stylize' in options &&
+    typeof options.stylize === 'function';
 
-  inspect(): string {
-    return `new BSONRegExp(${JSON.stringify(this.pattern)}, ${JSON.stringify(this.options)})`;
+  if (stylizeExists) {
+    return options.stylize as StylizeFunction;
+  } else {
+    return defaultInspect;
```

Comment:
super nit, generally if we early return inside an if then we don't write an `else` block